File: 20200701_programmers_sorting/h_index.py
"""
6 5 3 1 0
x x 3 1 0

6 5 4 3 2 1
x x x 3 2 1

6 3 1
x x 1

5 3 3
x x 3

5 4 4
x x x
"""


def solution(citations):
    answer = 0
    citations.sort(reverse=True)
    for i, c in enumerate(citations):
        answer = max(min(i+1, c), answer)
    return answer

File: 0701_정렬/Hindex.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    int answer = 0, low, high;
	vector<int> v;
	sort(citations.begin(), citations.end());
	
	int cit_sz = citations.size();
	int start = citations[0]; int end = citations[cit_sz - 1];
  if (start >= citations.size()) return citations.size(); //얘 안넣으면 tc9번 실패
  
	for (int i = start;i <= end;i++) { //i값이 h
		low = 0;
		high = 0;
		for (int j = 0;j<cit_sz;j++) {
			if (citations[j] < i) low++;
			else if (citations[j] >= i) high++;
		}
		if (high >= i) v.push_back(i);
	}

	answer = *max_element(v.begin(), v.end());
    return answer;
}
